src/main/controllers/features/common/checkSavedApplicationController.ts

```diff
@@ -0,0 +1,25 @@
+import { NextFunction, Request, Response } from "express";
+import * as config from "../../../config";
+import { validationResult } from "express-validator";
+import { formatValidationError, getPageProperties } from "../../../validation/validation";
+import { selectLang, addLangToUrl, getLocalesService, getLocaleInfo } from "../../../utils/localise";
+import { BASE_URL, SAVED_APPLICATION, TYPE_OF_BUSINESS, YOUR_FILINGS } from "../../../types/pageURL";
+import { Session } from "@companieshouse/node-session-handler";
+import { getSavedApplication, postAcspRegistration } from "../../../services/acspRegistrationService";
+import { createAndLogError, logger } from "../../../utils/logger";
+
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+    const session: Session = req.session as any as Session;
+    const savedApplication = await getSavedApplication(session, res.locals.userId);
+    const lang = selectLang(req.query.lang);
+    let nextPageUrl = "";
+    if (savedApplication.status === 404) {
+        nextPageUrl = addLangToUrl(BASE_URL + TYPE_OF_BUSINESS, lang);
+    } else if (savedApplication.status === 204) {
+        nextPageUrl = addLangToUrl(BASE_URL + SAVED_APPLICATION, lang);
+    } else {
+        logger.error(`internal server error: ` + savedApplication.status);
+    }
+
+    res.redirect(nextPageUrl);
```

Comment:
nextPageUrl will be null in case of else condition

Reply:
Addressed

Reply:
How will it handled on the UI, I can see error being logged and nothing will be rendered?